src/zenml/integrations/mlflow/steps/mlflow_registry.py

```diff
@@ -146,7 +146,11 @@ def mlflow_register_model_step(
         metadata.zenml_pipeline_run_uuid = pipeline_run_uuid
     if metadata.zenml_workspace is None:
         metadata.zenml_workspace = zenml_workspace
-    if metadata.model_extra.get("mlflow_run_id", None) is None:
+
+    # Check the model extra
+    if metadata.model_extra is None:
+        metadata.__pydantic_extra__ = {"mlflow_run_id": mlflow_run_id}
```

Comment:
Do we need to really use some pydantic internals here?
Shouldn't this just be `metadata.mlflow_run_id = mlflow_run_id`?
Or `setattr(metadata, "mlflow_run_id", mlflow_run_id)` if mypy complains about the first one?

Reply:
Done